### PR TITLE
Fix canceling animations.

### LINKED ISSUE #6111: [Bug, probably] Animation disposal does not dispose internal clock subscription
**Describe the bug**
After calling Dispose on Animation.Apply result it continues to change target value on each clock step.
I am trying to write test for animation cancellation (see https://github.com/AvaloniaUI/Avalonia/pull/6018), but it seems to not work as I expect. Maybe I expect wrong behavior. It does not seems to be a problem in real application, so might be mocks issue, or is it?

**To Reproduce**

```csharp
        [Fact]
        public void Dispose_Subscription_Should_Stop_Animation()
        {
            var keyframe1 = new KeyFrame()
            {
                Setters =
                {
                    new Setter(Border.WidthProperty, 200d),
                },
                Cue = new Cue(1d)
            };

            var keyframe2 = new KeyFrame()
            {
                Setters =
                {
                    new Setter(Border.WidthProperty, 100d),
                },
                Cue = new Cue(0d)
            };

            var animation = new Animation()
            {
                Duration = TimeSpan.FromSeconds(10),
                Delay = TimeSpan.FromSeconds(0),
                DelayBetweenIterations = TimeSpan.FromSeconds(0),
                IterationCount = new IterationCount(1),
                Children =
                {
                    keyframe2,
                    keyframe1
                }
            };

            var border = new Border()
            {
                Height = 100d,
                Width = 50d
            };
            var propertyChangedCount = 0;
            var animationCompletedCount = 0;
            border.PropertyChanged += (sender, e) =>
            {
                if (e.Property == Control.WidthProperty)
                {
                    propertyChangedCount++;
                }
            };

            var clock = new TestClock();
            var disposable = animation.Apply(border, clock, Observable.Return(true), () => animationCompletedCount++);

            Assert.Equal(0, propertyChangedCount);

            clock.Step(TimeSpan.FromSeconds(0));
            Assert.Equal(0, animationCompletedCount);
            Assert.Equal(1, propertyChangedCount);

            disposable.Dispose();

            // Clock ticks should be ignored after Dispose
            clock.Step(TimeSpan.FromSeconds(5));
            clock.Step(TimeSpan.FromSeconds(6));
            clock.Step(TimeSpan.FromSeconds(7));

            // On animation disposing (cancellation) on completed is not invoked (is it expected?)
            Assert.Equal(0, animationCompletedCount);
            // Initial property changed before cancellation + animation value removal = 2 changes.
            Assert.Equal(2, propertyChangedCount); // Fails here, actual value = 4, increasing after each additional clock.Step
        }
```

**Expected behavior**

**Desktop (please complete the following information):**

- OS: tests, probably a problem with mocks
- Version 0.10.6

### LINKED ISSUE #9387: Routing to old pages get blank content using RoutedViewHost
### Describe the bug 🐞

When trying to navigate to a page that was previously loaded, RoutedViewHost will become a blank page.

### Step to reproduce

1. Go to the next page
2. Then go to a page that was loaded
3. See a blank page
4. If click next and then navigate to that page again, the page will show up again

### Reproduction repository

_No response_

### Expected behavior

When navigate to a previous page, the view should show up normally

### Screenshots 🖼️

动画

### IDE

Visual Studio 2022

### Operating system

Windows

### Version

11, 22621.755

### AvaloniaUI Version

Avalonia 11.0 - Preview3

### ReactiveUI Version

18.3.1

### Additional information ℹ️

_No response_

### LINKED ISSUE #9514: Some items disappear when animating RoutedViewHost navigation
**Describe the bug**
I've noticed some items in View suddenly disappear when RoutedViewHost switching the page.

**To Reproduce**

1. Build and run attached project:
[BugRoutedViewHost.zip](https://github.com/AvaloniaUI/Avalonia/files/10066796/BugRoutedViewHost.zip)
2. After app starts, see "Some Caption" TextBlock and TabControl below are removed from page while it fades and switched to the Second page.
3. See all items from second page disappear too when it fades and navigates back to the Main page.
4. I'm not changing anything on page myself, it just executes the following method:

```
    private async void ReproduceBug()
    {
        await Task.Delay(3000);
        Router.Navigate.Execute(SecondPage);
        await Task.Delay(3000);
        Router.NavigateBack.Execute();
        await Task.Delay(3000);
        ReproduceBug();
    }
```

**Expected behavior**
I expect all items to fade evenly and page to not change its configuration while it's animating.

**Screenshots**
See the repro at the video:
https://user-images.githubusercontent.com/883587/203304501-b189afc9-42fa-4dc4-8dd7-143c5654bbdb.mp4

**Desktop (please complete the following information):**

- OS: Win 11 22H2 at 125% UI scale
- Version 11.0.0-preview.4

### LINKED ISSUE #9621: ViewModelViewHost, RoutedViewHost transitions are broken
**Describe the bug**
When using any of these controls, transitions became constantly fail to reach its final state.
For `CrossFade`, page is being partially or almost fully transparent.
For `PageSlide`, page is being bring in and out, so the view is not visible.

And it seems that TransitioningContentControl is not affected if you simply put it on a window, so it's probably has something to do with how reactiveUi controls triggers transition.

Just as a note, latest change in blame window shows some cancellation changes, but I believe it is not directly related, since it worked well around two months ago from now.

**To Reproduce**

reproduces very well if you use viewmodelviewhost

### PR DESCRIPTION
**Pull request by grokys**
## What does the pull request do?

As described in #6111, canceling an animation didn't do anything, which after the value store refactor in #8600 caused `RoutedViewHost` to stop working as described in #9514 (and maybe #9621).

The problem was that a `DisposeAnimationInstanceSubject` was being created but then just discarded and so was never disposed (which I assume is the purpose of this class, given its name). Make `Animator<T>.Apply` actually return the `DisposeAnimationInstanceSubject` so that it gets disposed when the animation gets canceled.

Because ControlCatalog doesn't have a dependency on `Avalonia.ReactiveUI` I had no way to test whether `RoutedViewHost` was fixed, so I also added a `ReactiveUIDemo` sample project which currently just contains a simple `RoutedViewHost` example for testing.

@jmacato you wrote this code, so I'd appreciate a 👀 .

## Fixed issues

Fixes #6111
Fixes #9514
Fixes #9621 
Might fix #9621 (need to check)
Might fix #9387 (need to check)

### CODE REVIEW

File: src/Avalonia.Base/Animation/Animators/Animator`1.cs [hunk 1]
```diff
@@ -39,7 +40,7 @@ public Animator()
                 VerifyConvertKeyFrames();
 
             var subject = new DisposeAnimationInstanceSubject<T>(this, animation, control, clock, onComplete);
-            return match.Subscribe(subject);
+            return new CompositeDisposable(match.Subscribe(subject), subject);
```
Comment (grokys) [inline src/Avalonia.Base/Animation/Animators/Animator`1.cs:43]: @jmacato this is the actual changed line in this PR.

**Review by jmacato — APPROVED** [review]
LGTM!

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027179-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-nightly/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027193-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-nightly/nuget/v3/index.json) [PRBUILDID]

**Pull request merged**